modules/carousel/jetpack-carousel.php

```diff
@@ -62,10 +66,14 @@ function init() {
 			 * @param array $this->prebuilt_widths Array of default widths.
 			 */
 			$this->prebuilt_widths = apply_filters( 'jp_carousel_widths', $this->prebuilt_widths );
-			add_filter( 'post_gallery', array( $this, 'enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
+			add_filter( 'post_gallery', array( $this, 'check_and_enqueue_assets' ), 1000, 2 ); // load later than other callbacks hooked it
 			add_filter( 'post_gallery', array( $this, 'set_in_gallery' ), -1000 );
 			add_filter( 'gallery_style', array( $this, 'add_data_to_container' ) );
 			add_filter( 'wp_get_attachment_image_attributes', array( $this, 'add_data_to_images' ), 10, 2 );
+			if ( $this->single_image_gallery_enabled ) {
+				add_filter( 'the_content', array( $this, 'add_data_to_single_images' ) );
+				$this->enqueue_assets();
```

Comment:
Hmmm... Guess this condition needs to be changed to `$this->single_image_gallery_enabled && no_other_plugin_is_handling_gallery_shortcode`. Will correct it.